### PR TITLE
Don't use separate variables for tracking number of trinkets and crewmates

### PR DESCRIPTION
**Pull request by InfoTeddy**
### Summary

This changes the variables `game.trinkets` and `game.crewmates` to be functions which count the number of collected trinkets and crewmates automatically, so we don't have to keep track of the amount of collected trinkets and crewmates separately. It also does the same for `ed.numtrinkets` and `ed.numcrewmates` to keep track of how many trinkets and crewmates there are in the level overall. `map.customtrinkets` and `map.customcrewmates` have been removed due to simply being duplicates of `ed.numtrinkets` and `ed.numcrewmates`.

Also, I cleaned up unused or useless variables.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [X] My changes may be used in a future commercial release of VVVVVV (for
  example, a 2.3 update on Steam for Windows/macOS/Linux)
- [X] I will be credited in a `CONTRIBUTORS` file for all of said releases, but
  will NOT be compensated for these changes

### CODE REVIEW

File: desktop_version/src/Game.cpp [hunk 1]
```diff
@@ -7606,20 +7606,7 @@ void Game::swnpenalty()
...
-    if (crewstats[2])
-        temp++;
-    if (crewstats[3])
-        temp++;
-    if (crewstats[4])
-        temp++;
-    if (crewstats[5])
-        temp++;
-    return temp;
+    return std::count(crewstats.begin(), crewstats.end(), true);
```
Comment (flibitijibibo) [inline desktop_version/src/Game.cpp:7607]: Would a `for` loop be more appropriate here? Again, the less STL the better.

File: desktop_version/src/Game.cpp [hunk 2]
```diff
@@ -7654,3 +7638,8 @@ void Game::resetgameclock()
     hours = 0;
     timerStartTime = SDL_GetTicks();
 }
+
+int Game::trinkets()
+{
+    return std::count(obj.collect.begin(), obj.collect.end(), 1);
```
[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.cpp]: What do you mean by `size()`? You mean using `std::vector::size()`? (Which doesn't seem right if that's what you mean...)

[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.cpp]: I could always just replace it with a for-loop, but your question confuses me.

**Comment by flibitijibibo** [comment]
A loop is also okay - that does put more emphasis on the “we turned a variable into a function” issue though.

**Comment by InfoTeddy** [comment]
What exactly did you mean by "Is it possible for this to be `size()`?", though?

File: desktop_version/src/Game.h [hunk 3]
```diff
@@ -247,7 +247,7 @@ class Game
 
     int deathseq, lifeseq;
 
-    int coins, trinkets, crewmates, trinkencollect;
+    int coins, trinkets(), crewmates, trinkencollect;
```
[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.h]: Fixed in a0e57bacb37ceb3aa6e9596b644003d441666066.

File: desktop_version/src/Game.h [hunk 4]
```diff
@@ -247,7 +247,7 @@ class Game
 
     int deathseq, lifeseq;
 
-    int coins, trinkets(), crewmates, trinkencollect;
+    int coins, trinkets(), crewmates(), trinkencollect;
```
[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.h]: Fixed in a0e57bacb37ceb3aa6e9596b644003d441666066.

File: desktop_version/src/Game.cpp (hunk 1 above)
Comment (InfoTeddy) [reply desktop_version/src/Game.cpp]: Fixed in a7a81a45f1c02276dff645a187265352c3b71b18.

**Comment by InfoTeddy** [comment]
Did you mean the `size()` in a statement like `for (size_t i = 0; i < myvector.size(); i++)`? Makes sense if so.

**Comment by flibitijibibo** [comment]
Really I just wanted an explicitly-written function that shows exactly what's being counted and how, without STL obscuring that work away from the reader. In my head I saw that line and thought that it could have been a basic size value but in reality it does indeed have to be a loop.

**Comment by InfoTeddy** [comment]
Makes sense, then.

File: desktop_version/src/Game.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.cpp]: Fixed in 2deba8b009025963d35df6ec8a52a677cedfc972.

File: desktop_version/src/Game.cpp [hunk 5]
```diff
@@ -7643,3 +7638,8 @@ int Game::trinkets()
 {
     return std::count(obj.collect.begin(), obj.collect.end(), 1);
 }
+
+int Game::crewmates()
+{
+    return std::count(obj.customcollect.begin(), obj.customcollect.end(), 1);
```
[the remark this answers is not in the document]
Comment (InfoTeddy) [reply desktop_version/src/Game.cpp]: Fixed in f4a2d28102b9f53997c50c3a44ebc8b09664ab29.

**Pull request merged**